src/crimson/osd/pg.cc

```diff
@@ -634,7 +654,11 @@ seastar::future<Ref<MOSDOpReply>> PG::do_osd_ops(
       obc->obs.oi.soid,
       ceph_osd_op_name(osd_op.op.op));
     return ox->execute_op(osd_op);
-  }).safe_then([this, obc, m, ox = ox.get(), &op_info] {
+  }).handle_error(
+      PGBackend::disk_or_crc_caused_eio::handle([&osd_op] {
+        return rep_repair_primary_object(oid, osd_op);
```

Comment:
Not sure what to return here. Looks like `ClientRequest::process_op()` is expecting an MOSDOpReply, should I format a dedicated reply that indicates that the object was recovered? If so then we'll need to add special handling for that specific reply (because we might want to try restarting the op entirely if the object is not missing anymore, or continue from where we left off). What do you think? @tchaikov @xuxuehan

Reply:
@AmnonHanuhov i think this question is already addressed by the answers in your shared google document of "EIO handling".  if you are still confused, probably we can discuss this in details in next crimson standup or a time slot which works for you and (@athanatos and/or @rzarzynski and/or me). i am quoting Sam's reply here

> I think you just need to kick off an UrgentRecovery asn in process_op and wait for it to resolve.